src/sentry/search/events/builder.py

```diff
@@ -1645,12 +1660,133 @@ def __init__(
+        return conditions
+
+    def resolve_groupby(self, groupby_columns: Optional[List[str]] = None) -> List[SelectType]:
+        if groupby_columns is None:
+            groupby_columns = []
+        return list(
+            {col for col in self.columns if col.name in self.required_select_groupby}
+            | {self.resolve_column(column) for column in groupby_columns}
+        )
+
+    def resolve_granularity(self) -> Optional[Granularity]:
+        if self._granularity is None:
+            return None
+        return Granularity(self._granularity)
+
+    def resolve_query(
+        self,
+        query: Optional[str] = None,
+        use_aggregate_conditions: bool = False,
+        selected_columns: Optional[List[str]] = None,
+        groupby_columns: Optional[List[str]] = None,
+        equations: Optional[List[str]] = None,
+        orderby: Optional[List[str]] = None,
+    ) -> None:
+        super().resolve_query(
+            query, use_aggregate_conditions, selected_columns, groupby_columns, equations, orderby
+        )
+        with sentry_sdk.start_span(op="QueryBuilder", description="resolve_granularity"):
+            # Needs to happen before params and after time conditions since granularity can change start&end
+            self.granularity = self.resolve_granularity()
+
+    def get_snql_query(self) -> Request:
+        self.validate_having_clause()
+
+        return Request(
+            dataset=self.dataset.value,
+            app_id="default",
+            query=Query(
+                match=Entity(self.dataset.value, sample=self.sample_rate),
+                select=self.columns,
+                array_join=self.array_join,
+                where=self.where,
+                having=self.having,
+                groupby=self.groupby,
+                orderby=self.orderby,
+                limit=self.limit,
+                offset=self.offset,
+                granularity=self.granularity,
+                limitby=self.limitby,
+            ),
+            flags=Flags(turbo=self.turbo),
+        )
+
+    def _default_filter_converter(self, search_filter: SearchFilter) -> Optional[WhereType]:
+        name = search_filter.key.name
+        if name in self.condition_fields or name in self._extra_condition_fields:
+            return super()._default_filter_converter(search_filter)
+        raise InvalidSearchQuery(f"Invalid search filter: {name}")
+
+
+class TimeseriesSessionsV2QueryBuilder(SessionsV2QueryBuilder):
+    time_column = "bucketed_started"
+
+    def __init__(
+        self,
+        *args: Any,
+        extra_conditions: Optional[List[WhereType]] = None,
+        **kwargs: Any,
+    ):
+        self._extra_conditions = extra_conditions
+        super().__init__(*args, **kwargs)
+
+    def get_snql_query(self) -> Request:
+        self.validate_having_clause()
+
+        where = self.where
+        if self._extra_conditions:
+            where += self._extra_conditions
```

Comment:
For both builders, could we move _extra_conditions to be something handled by the caller instead of an extra parameter?
ie.
```python
x = TimeseriesSessionsV2QueryBuilder(...)
x.where.append(...)
x.get_snql_query()
```

Reply:
Sure 

What do you mean both builders though, this is only in the Timeseries builder?